SOSS/dms/soss_solver.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# TODO I don't much like the way the rotation point is handled.
+# TODO Theoretically it could be removed entirely, but the way coords and image
+# TODO handle it are different by default (lower-left vs center).
+
+from astropy.io import fits
+import numpy as np
+from scipy.ndimage import shift, rotate
+from scipy.optimize import minimize
+import warnings
+
+from .soss_centroids import get_soss_centroids
+
+
+def transform_coords(angle, xshift, yshift, xpix, ypix, cenx=1024, ceny=50):
+    """Apply a rotation and shift to the trace centroids positions. This
+    assumes that the trace centroids are already in the CV3 coordinate system.
+
+    :param angle: The angle by which to rotate the coordinates, in degrees.
+    :param xshift: The shift to apply to the x-coordinates after rotating.
+    :param yshift: The shift to apply to the y-coordinates after rotating.
+    :param xpix: The x-coordinates to be transformed.
+    :param ypix: The y-coordinates to be transformed.
+    :param cenx: The x-coordinate around which to rotate.
+    :param ceny: The y-coordinate around which to rotate.
+
+    :type angle: float
+    :type xshift: float
+    :type yshift: float
+    :type xpix: array[float]
+    :type ypix: array[float]
+    :type cenx: The x-coordinate around which to apply the rotation.
+    :type ceny: The y-coordinate around which to apply the rotation.
+
+    :returns: xrot, yrot - The rotated and shifted coordinates.
+    :rtype: Tuple(array[float], array[float])
+    """
+
+    # Convert to numpy arrays.
+    xpix = np.atleast_1d(xpix)
+    ypix = np.atleast_1d(ypix)
+
+    # Required rotation in the detector frame to match the data.
+    t = np.deg2rad(angle)
```

Comment:
Just a comment to avoid one letter variables - they are hard to search for - if there is any name for `t` and `R` it would be preferable.

Reply:
fixed in commit 8feda64